src/jdk.naming.dns/share/classes/com/sun/jndi/dns/DnsClient.java

```diff
@@ -159,6 +181,10 @@ protected void finalize() {
     private Object queuesLock = new Object();
 
     public void close() {
+        try {
+            udpChannelSelector.close();
```

Comment:
Do you think we should now maintain a `closed` boolean flag in this class to keep track of whether this underlying selector has been closed? 

The javadoc of `Selector.close()` states that any subsequent use of the selector will throw a `ClosedSelectorException`. A `ClosedSelectorException` is a `RuntimeException`, so if a closed `DnsClient` gets used (for example a `query()` gets triggered) then from what I can see it will end up propagating this `ClosedSelectorException` out of these class methods instead of the declared compile time exceptions.

Maintaining a `closed` flag could allow us to use that flag to check in these methods (where we use the selector) and throw a more appropriate exception.

Reply:
Good point - planning to address the closed selector in the `blockingReceive` method by calling `udpChannelSelector.isOpen()`.

Reply:
Addressed it by catching `ClosedSelectorException` and wrapping it into JNDI's `CommunicationException` in 55dd0a4a